django/core/management/commands/shell.py

```diff
@@ -111,9 +114,35 @@ def python(self, options):
         # Start the interactive interpreter.
         code.interact(local=imported_objects)
 
+    def get_apps_and_models(self):
+        apps_and_models = {}
+        for app_conf in apps.get_app_configs():
+            if app_conf.models_module:
+                apps_and_models[app_conf.models_module] = app_conf.get_models()
+        return apps_and_models
+
+    def get_namespace(self, options):
+        apps_and_models = self.get_apps_and_models()
+        imported_objects = {}
+        for app_module, app_models in reversed(apps_and_models.items()):
+            if not app_models:
+                continue
+            for model in app_models:
+                if model.__module__:
+                    imported_objects[model.__name__] = model
+        if not options["command"]:
```

Comment:
I think we should also avoid the message when running code from stdin too. It may be better to have the callers pass a `silent` flag rather than check `options` here.